src/views/BookingPage.vue

```diff
@@ -0,0 +1,129 @@
+import BookingHeader from "@/components/bookings/BookingHeader.vue";
+import SeatingGrid from "@/components/bookings/SeatingGrid.vue";
+import BaseButton from "@/components/base/BaseButton.vue";
+import BaseSelect from "@/components/base/BaseSelect.vue";
+import BaseInput from "@/components/base/BaseInput.vue";
+import { ticketIds } from "@/api/ticketID";
+const bookingStore = useBookingStore();
+
+const selectedTab = ref("Choose seats");
+
+const headerTitle = computed(() => {
+  return `Choose your ${
+    selectedTab.value === "Choose seats" ? "seats" : "tickets"
+  }`;
+});
+const selectOptions = computed(() => {
+  return ticketIds.map((ticket) => {
+    return {
+      id: ticket.id,
+      title: `${ticket.type} - $${ticket.price.slice(0, -2)}`,
+    };
+  });
+});
+
+const handleChooseSeats = () => {
+  if (bookingStore.selectedTickets.length > 0) {
+    selectedTab.value = "Book tickets";
+  }
+};
+const handleBookTickets = async () => {
+  const selectionData = {
+    seance_id: bookingStore.selectedReservation.id,
+    tickets: bookingStore.selectedTickets,
+  };
+  postReservation(selectionData);
+};
+</script>
+
+<template>
+  <div>
+    <BookingTabs @selected="(val) => (selectedTab = val)" />
+    <h1>{{ headerTitle }}</h1>
+    <BookingHeader
+      v-if="bookingStore.selectedReservation"
+      :movie_id="bookingStore.selectedReservation.movie"
+    />
+    <template v-if="selectedTab === 'Choose seats'">
+      <SeatingGrid />
+      <div class="reservations__controls reservations__controls--choose">
+        <BaseButton
+          :disabled="bookingStore.selectedTickets.length === 0"
+          size="large"
+          type="secondary"
+          modifier="outlined"
+          @click="handleChooseSeats"
+        >
+          Choose seats
+        </BaseButton>
+      </div>
+    </template>
+    <template v-else-if="selectedTab === 'Book tickets'">
+      <div>
+        <div
+          class="reservations__tickets-list"
+          v-for="(ticket, index) in bookingStore.selectedTickets"
+          :key="ticket.seat"
+        >
+          <BaseInput
+            :disabled="true"
+            v-model="bookingStore.selectedTickets[index].seat"
+          >
+            Seat
+          </BaseInput>
+          <BaseSelect
+            :options="selectOptions"
+            v-model="bookingStore.selectedTickets[index].ticket_type_id"
+            >Ticket type</BaseSelect
+          >
+        </div>
+      </div>
+      <div class="reservations__controls reservations__controls--book">
+        <BaseButton
+          size="large"
+          type="secondary"
+          modifier="outlined"
+          @click="selectedTab = 'Choose seats'"
+        >
+          Go back
+        </BaseButton>
+        <BaseButton
+          size="large"
+          type="secondary"
+          modifier="outlined"
+          @click="handleBookTickets"
+        >
+          Book tickets
+        </BaseButton>
+      </div>
+    </template>
+  </div>
+</template>
```

Comment:
I do like how you structured the logic in this template, good job 😁

Reply:
Thank you